os/arch/arm/src/amebasmart/amebasmart_i2c.c

```diff
@@ -1159,11 +1164,17 @@ FAR struct i2c_dev_s *up_i2cinitialize(int port)
 
 	/* Get I2C private structure */
 	if (port == 0) {
+#ifdef CONFIG_AMEBASMART_I2C0
 		priv = (struct amebasmart_i2c_priv_s *)&amebasmart_i2c0_priv;
+#endif
 	} else if (port == 1) {
+#ifdef CONFIG_AMEBASMART_I2C1
 		priv = (struct amebasmart_i2c_priv_s *)&amebasmart_i2c1_priv;
+#endif
 	} else if (port == 2) {
+#ifdef CONFIG_AMEBASMART_I2C2
 		priv = (struct amebasmart_i2c_priv_s *)&amebasmart_i2c2_priv;
+#endif
 	} else {
 		printf("Please select I2CO, I2C1 or I2C2 bus\n");
 		return NULL;
```

Comment:
You should check this block again.
If we disable I2C1 and call port 1 in upper layer, the `priv` is null by line 1162.
Then, line 1184 does null dereference. (priv->i2c_object)